src/backtrace/dbghelp64.rs

```diff
@@ -89,8 +89,9 @@ pub unsafe fn trace(cb: &mut dyn FnMut(&super::Frame) -> bool) {
     let mut context = core::mem::zeroed::<MyContext>();
     RtlCaptureContext(&mut context.0);
 
-    // Call `RtlVirtualUnwind` to find the previous stack frame, walking until we hit ip = 0.
-    while context.ip() != 0 {
+    // Call `RtlVirtualUnwind` to find the previous stack frame, walking until we hit ip = 0
+    // or we are no longer making progress.
+    loop {
         // The base address of the module containing the function will be stored here
         // when RtlLookupFunctionEntry returns successfully.
         let mut base = 0;
```

Comment:
Hmm. Interesting. I guess we `break` when we hit `RtlLookupFunctionEntry` returning `is_null` instead of first calling our callback because we don't expect our caller to coherently deal with a frame lacking a `base_address`. Yet we sure *could* decide to pass that obligation on: we're calling a safe callback, not an unsafe one, so if they deref a pointer we gave them or otherwise make bad assumptions that cause UB, that's technically "on them". It *is* probably rude, though, when we could simply swallow the error and stop here.

This should probably have a comment to that effect.

Reply:
If `RtlLookupFunctionEntry` is returning null then no function entry could be found: this usually indicates that there's something wrong with the stack/process (corrupted, binaries have been unloaded, etc.)